### DISCUSSION TITLE
Documentation confusion re: association proxy

### DISCUSSION THREAD

**Discussion by BackSeat** [sqlalchemy/sqlalchemy#8212] [Usage Questions]
I'm not sure whether this is a bug in the docs or a bug in my brain.

In [Simplifying Scalar Collections](https://docs.sqlalchemy.org/en/20/orm/extensions/associationproxy.html#simplifying-scalar-collections), an example table is shown:

```
userkeywords_table = Table(
    "userkeywords",
```
Later, in  [Simplifying Association Objects](https://docs.sqlalchemy.org/en/20/orm/extensions/associationproxy.html#simplifying-association-objects) section, it says "Below, we illustrate a new mapping which introduces the `UserKeyword` class, which is mapped to the `userkeywords` table illustrated earlier." But it's not. The `UserKeyword` class maps to the `user_keyword` table:

```
class UserKeyword(Base):
    __tablename__ = "user_keyword"
```

It may sound pedantic, but the use of the Association Proxy is not the simplest of concepts, and I spent some time trying to ensure that  the change in table names was not just my misunderstanding. My conclusion is that it's a bug in the doco.

It gets more confusing. In the `User` class there is:

```
keywords = association_proxy("user_keywords", "keyword")
```

But the association table is `user_keyword`. There's a backreference to `user_keywords` in the `UserKeyword` class, but I don't understand why the association object is linked to the backreference rather than the table.

So...I kind of understand it, but I think the documentation is confusing and possibly even wrong.

**Comment by CaselIT** [comment]
Hi,

I think that the table could be called the same to make the docs clearer. Feel free to open a PR to propose a change. That part of the documentation is written here https://github.com/sqlalchemy/sqlalchemy/blob/main/doc/build/orm/extensions/associationproxy.rst

**Comment by gordthompson** [comment]
https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/3981
